<font color="blue">4. La longitud de tornillo sigue un distribución normal con media m y y desviación típica 0.3. ¿Cuál debe ser el tamaño mínimo de la muestra para que la probabilidad de que la media muestral difiera de la media en (a lo sumo) 0.2 cm sea de 0.99?</font>


<h3>Solución. </h3>  Si la muestra tiene tamaño $n$, entonces $$\overline{X}\sim N(m,\frac{0.3}{\sqrt{n}}).$$
Por tanto, 
$$\frac{\overline{X}-m}{\frac{0.3}{\sqrt{n}}}\sim N(1,0).$$
Además, 
$$\left|\overline{X}-m\right|<0.2 \Leftrightarrow \left|\frac{\overline{X}-m}{\frac{0.3}{\sqrt{n}}}\right|< \frac{0.2}{\frac{0.3}{\sqrt{n}}} = \frac{0.2\sqrt{n}}{0.3} = \frac{2}{3}\sqrt{n}$$
De modo que 
$$ P\left(\left|\overline{X}-m\right|<0.2 \right) = 0.99 \Leftrightarrow P\left(\left|\frac{\overline{X}-m}{\frac{0.3}{\sqrt{n}}}\right|< \frac{2}{3}\sqrt{n}\right) = 0.99$$
Como la distribución normal $N(0,1)$ es simétrica respecto del origen, podemos concluir que el area que hay en la cola a partir de $\frac{2}{3}\sqrt{n}$ es $0.005$, por lo que $\frac{2}{3}\sqrt{n}$ debe coincidir con el percentil $0.995$


In [4]:
from scipy.stats import norm
X=norm(0,1)
X.ppf(0.995)

2.5758293035489004



De modo que $$\frac{2}{3}\sqrt{n}=2.575829$$
Equivalentemente, 
$$\left(\frac{3}{2}2.575829\right)^2=n$$


In [6]:
n=(3/2*X.ppf(0.995))**2
n

14.92851735229773

El tamaño de la muestra debe ser $n=15$. De hecho, cualquier valor entero superior a 14 nos sirve. 

<font color="blue"> 5. Se sabe que el 45% de una población de votantes está a favor del partido A y el resto a favor del partido B.

*    Se hace una encuesta a 100 personas, ¿cuál es la probabilidad de que esta encuesta de como
vencedor al partido A? ¿Cuál es la probabilidad de que la proporción de votantes del partido A en esta muestra esté entre 44% y 46%?
*     Se hace una encuesta a 1000 personas, ¿cuál es la probabilidad de que esta encuesta de como
vencedor al partido A? Comparar con el resultado obtenido en el apartado anterior.</font> 


<h3>Solución. </h3> Sabemos que la proporción muestral $\hat{p}$ satisface $$\hat{p}\sim N(p,\sqrt{\frac{p(1-p)}{n}})$$



In [1]:
p=0.45
n=100
(p*(1-p)/n)**0.5

0.049749371855331


Por tanto, $\hat{p}\sim N(0.45, 0.04974937)$ y nos piden:

*  La probabilidad de que la encuesta de como vencedor a $A$.

Esto coincide con $P[\hat{p}>0.5]=1-P[\hat{p}\leq 0.5]$, que es:



In [4]:
from scipy.stats import norm


0.15743931164448965

* La probabilidad de que proporción de votantes de $A$ esté entre 0.44 y 0.46.

Esto es, queremos conocer $P(0.44\leq \hat{p}\leq 0.46)$


In [5]:
X.cdf(0.46)-X.cdf(0.44)

0.15930734555606652

* Se hace una encuesta a 1000 personas, ¿cuál es la probabilidad de que esta encuesta de como
vencedor al partido A? Comparar con el resultado obtenido en el apartado anterior.

En este caso $n=1000$ de modo que: 
 



In [6]:
p=0.45
n=1000
(p*(1-p)/n)**0.5

0.015732132722552274


$$\hat{p}\sim N(0.45, 0.01573213)$$
y ahora $$P[\hat{p}>0.5]=1-P[\hat{p}\leq 0.5]$$ vale:


In [7]:
X=norm(0.45, 0.01573213)
1-X.cdf(0.5)

0.0007409389817195633


Es evidente que cuanto mayor sea el tamaño de la muestra, más preciso será el resultado de la predicción estimada por $\hat{p}$ y, por tanto, más improbable será que la encuesta arroje un resultado erróneo.  
